src/wp-includes/https-detection.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * HTTPS detection functions.
+ *
+ * @package WordPress
+ * @since 5.7.0
+ */
+
+/**
+ * Checks whether the website is using HTTPS.
+ *
+ * This is based on whether the home and site URL are using HTTPS.
+ *
+ * @since 5.7.0
+ *
+ * @return bool True if using HTTPS, false otherwise.
+ */
+function wp_is_using_https() {
+	if ( 'https' !== wp_parse_url( home_url(), PHP_URL_SCHEME ) ) {
+		return false;
+	}
+
+	// Use direct option access for 'siteurl' and manually run the 'site_url'
+	// filter because site_url() will adjust the scheme based on what the
+	// current request is using.
+	/** This filter is documented in wp-includes/link-template.php */
+	$site_url = apply_filters( 'site_url', get_option( 'siteurl' ), '', null, null );
+
+	if ( 'https' !== wp_parse_url( $site_url, PHP_URL_SCHEME ) ) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Checks whether HTTPS is supported for the server and domain.
+ *
+ * @since 5.7.0
+ *
+ * @return bool True if HTTPS is supported, false otherwise.
+ */
+function wp_is_https_supported() {
+	$https_detection_errors = get_option( 'https_detection_errors' );
+
+	// If option has never been set by the Cron hook before, run it on-the-fly as fallback.
+	if ( false === $https_detection_errors ) {
+		wp_update_https_detection_errors();
+
+		$https_detection_errors = get_option( 'https_detection_errors' );
+	}
+
+	// If there are no detection errors, HTTPS is supported.
+	return empty( $https_detection_errors );
+}
+
+/**
+ * Runs a remote HTTPS request to detect whether HTTPS supported, and stores potential errors.
+ *
+ * This internal function is called by a regular Cron hook to ensure HTTPS support is detected and maintained.
+ *
+ * @since 5.7.0
+ * @access private
+ */
+function wp_update_https_detection_errors() {
```

Comment:
Per above, this could be:

```suggestion
 * @access private
 * @param bool|string $use_cached Whether to use cached value. (Empty string may be supplied due to use with add_action() for WP Cron.)
 */
function wp_update_https_detection_errors( $use_cached = '' ) {
	if ( ! empty( $use_cached ) ) {
		$errors = get_option( 'https_detection_errors', false );
		if ( false !== $errors ) {
			return $errors;
		}
	}
```